datasets/Wolters et al. (2018)/process_Wolters_dataset.ipynb

```diff
@@ -0,0 +1,317 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import pandas as pd\n",
+    "from pathlib import Path\n",
+    "from RISparser import read, TAG_KEY_MAPPING, LIST_TYPE_TAGS\n",
+    "\n",
+    "# visualization\n",
+    "import matplotlib.pyplot as plt\n",
+    "from wordcloud import WordCloud, STOPWORDS"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Read files from OSF"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url_all = \"https://osf.io/a26sz/download\"\n",
+    "url_included = \"https://osf.io/hy8qe/download\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "list_keys = [TAG_KEY_MAPPING[k] for k in LIST_TYPE_TAGS]\n",
+    "\n",
+    "def read_ris_to_df(url):\n",
+    "    \"\"\"Read RIS and return pandas DataFrame\"\"\"\n",
+    "    \n",
+    "    # download data and split into lines\n",
+    "    r = requests.get(url)\n",
+    "    r.encoding = \"utf-8-sig\"\n",
+    "    lines = r.text.split(\"\\r\\n\")\n",
+    "    \n",
+    "    # merge the field with multiple values\n",
+    "    items = []\n",
+    "    for item in read(lines):\n",
+    "        for k, v in item.items():\n",
+    "            if k in list_keys and item[k] is not None:\n",
+    "                item[k] = \";\".join(item[k])\n",
+    "        items.append(item)\n",
+    "\n",
+    "    return pd.DataFrame(items)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_all = read_ris_to_df(url_all)\n",
+    "df_included = read_ris_to_df(url_included)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Clean datasets"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# retain useful columns\n",
+    "use_col = [\"title\", \"abstract\", \"keywords\", \"authors\", \"year\", \"doi\"]\n",
```

Comment:
Is it possible to keep the ID column from the original data as well?

Reply:
Yes. Is there any other useful variable that should be kept? I'm working on standardizing all datasets in this repository. There are some inconsistencies in the process of datasets.

Reply:
I think the most important ones are there. Maybe in the future it would be interesting to keep the LA tag (for language) as well? Regarding the stop words for feature extraction (right @J535D165 @qubixes?) Another one could be the DB tag (for the search database). 

However, I'd expect there to be a lot of missing data for both tags, maybe it is not feasible to standardize this for all datsets..